### DISCUSSION TITLE
EPERM error

### DISCUSSION THREAD

**Discussion by dimoff66** [prisma/prisma#6376] [Q&A]
I'm moving my nodejs prisma project to production

I have a mysql database called **dimoff66_tokens**, user with the same name and password **Master_77**

When i get an access via mysql2 i type 

```
const connection = mysql.createConnection({
  host: 'localhost',
  user: 'dimoff66_tokens',
  database: 'dimoff66_tokens',
  password: 'Master_77'
});
```

and it works fine

For prisma i type

```
datasource db {
  provider = "mysql"
  url      = "mysql://dimoff66_tokens:Master_77@localhost:3306/dimoff66_tokens"
}
```

and making simple mysql query like prisma.exchange.findMany()

i get an error **{"code":"EPERM","clientVersion":"2.20.1"}**

What is the cause of it and how can i solve it?

**Comment by janpio** [comment]
Can you show the complete error you are getting? Does not sound like this is related to the connection string at all.
Does `npx prisma introspect` work?

**Reply by dimoff66** [reply]
> Can you show the complete error you are getting? Does not sound like this is related to the connection string at all.
> Does `npx prisma introspect` work?

Yes, it is not related
introspect and db push work well

my code in controller is

```
try {
  const data = await prisma.exchange.findMany()
  return { data }
}

catch (e) {
  return { error: JSON.stringify(e)}
}
```

and result i'm getting is
_"error":"{\"code\":\"EPERM\",\"clientVersion\":\"2.20.1\"}"_

No adiitional information

And it happens with any prisma model and its method on ubuntu server i rent

**Reply by janpio** [reply]
> Yes, it is not related

Yes it is, as it shows me your database connection string is correct and Prisma can connect successfully.



> ```
> try {
>   const data = await prisma.exchange.findMany()
>   return { data }
> }
> 
> catch (e) {
>   return { error: JSON.stringify(e)}
> }
> ```
> 
> 
> and result i'm getting is
> 
> _"error":"{"code":"EPERM","clientVersion":"2.20.1"}"_

Can you remove the try/catch block and let it crash to get a stacktrace or similar?

**Reply by dimoff66** [reply]
> Can you remove the try/catch block and let it crash to get a stacktrace or similar?

The problem is it happens on server and i'm not experienced in distant server debugging, when i remove it - all i get in browser is  "internal server error"... But i will try to find ways
